lib/Runtime/Language/AsmJsByteCodeGenerator.cpp

```diff
@@ -426,18 +426,22 @@ namespace Js
         // Emit a function body. Only explicit returns and the implicit "undef" at the bottom
         // get copied to the return register.
 
+        ParseNode *stmt = nullptr;
         while (varStmts->nop == knopList)
         {
-            ParseNode *stmt = ParserWrapper::GetBinaryLeft(varStmts);
+            stmt = ParserWrapper::GetBinaryLeft(varStmts);
             EmitTopLevelStatement( stmt );
             varStmts = ParserWrapper::GetBinaryRight(varStmts);
         }
         Assert(!varStmts->CapturesSyms());
 
         // if last statement isn't return, type must be void
-        if (varStmts->nop != knopReturn)
+        if (!stmt || stmt->nop != knopReturn)
         {
```

Comment:
Shouldn't we still check the right node of the binary statement ? (varStmts->nop)